README.md

```diff
@@ -1,18 +1,55 @@
 # Puppet::Modulebuilder
 
+## Table of Contents
+
+1. [Overview - What is Puppet::Modulebuilder?](#overview)
+2. [Description - What does the gem do?](#description)
+3. [Usage - How can the gem be used?](#usage)
+4. [Testing - How to test changes to the gem?](#contributing)
+5. [Contributing - How to contribute to the gem?](#contributing)
+6. [Development - How to release changes to the gem?](#development)
+
+## Overview
+
 The `puppet-modulebuilder` gem contains the reference implementation for building Puppet modules from source.
 
+## Description
+
+The purpose of this tool is to take a given local module directory and compile it into a `.tar` file, known as the `tarball`, that can then be installed directly by Puppet on a target machine or uploaded onto the [Puppet Forge](https://forge.puppet.com/) so that it can be accessed publicly.
+
+As part of this process any non-deliverable aspects of the module, parts of it related to the modules development or testing for example, are stripped away leaving only the documentation and the puppet/ruby code that is needed for the module to function.
```

Comment:
Should it describe which files it reads here? Like `.gitignore`, `.pmtignore`, `.pdkignore` and in which order?

Reply:
Created a new PR with an update: https://github.com/puppetlabs/puppet-modulebuilder/pull/62